lib/charliecloud.py

```diff
@@ -55,6 +56,15 @@
 
 ## Globals ##
 
+# Map common uname machine type to manifest platform architecture.
+ARCH_MAP = {'aarch32': {'architecture': 'arm',   'variant': 'v7'},
+            'aarch64': {'architecture': 'arm64', 'variant': 'v8'},
+            'armv5l':  {'architecture': 'arm',   'variant': 'v5'},
+            'armv6l':  {'architecture': 'arm',   'variant': 'v6'},
+            'armv7l':  {'architecture': 'arm',   'variant': 'v7'},
+            'armv8l':  {'architecture': 'arm64', 'variant': 'v8'},
+            'x86_64':  {'architecture': 'amd64'}}
```

Comment:
What about tuples instead of subdictionaries?

Also, if `variant` must be present, then there will be fewer code paths.

Use double quotes.

Reply:
I can fiddle with tuples instead of dictionaries here.

Variant is optional.